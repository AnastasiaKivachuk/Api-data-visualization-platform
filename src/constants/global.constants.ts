export const DATE_FORMAT = {
    DATE_TIME_NUMBER: 'DD.MM.YYYY, HH:mm A',
    DATE_WORD: 'MMM DD, YYYY',
};