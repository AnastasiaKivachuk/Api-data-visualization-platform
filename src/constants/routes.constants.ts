export const HOME_ROUTE = '/';

export const USERS_ROUTE = '/users';
export const USER_ROUTE = '/users/[id]';

export const LAUNCHES_ROUTE = '/launches';
export const LAUNCH_ROUTE = '/launches/[id]';