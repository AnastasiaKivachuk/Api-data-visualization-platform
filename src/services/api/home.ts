import { gql } from '@apollo/client';
import client from '../../../apollo-client';

export const getHome = () => client.query({
  query: gql`
        query HomeScheme {
            company {
                ceo
                employees
                founded
                founder
                links {
                    elon_twitter
                    website
                }
                name
                summary
            }
            ships {
                id
                image
                model
                name
                url
                year_built
            }
            rockets {
                wikipedia
                name
                id
                description
            }
        }
    `,
});
