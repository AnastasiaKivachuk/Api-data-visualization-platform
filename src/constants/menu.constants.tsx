import {HOME_ROUTE, LAUNCHES_ROUTE, USERS_ROUTE} from './routes.constants';
import GroupIcon from '@mui/icons-material/Group';
import HomeIcon from '@mui/icons-material/Home';
import RocketLaunchIcon from '@mui/icons-material/RocketLaunch';
import React from "react";

export const APP_BAR_ITEMS = [
    {
        link: HOME_ROUTE,
        text: '',
        icon: <HomeIcon/>
    },
    {
        link: USERS_ROUTE,
        text: 'Users',
        icon: <GroupIcon/>
    },
    {
        text: 'Launches',
        link: LAUNCHES_ROUTE,
        icon: <RocketLaunchIcon/>
    }
]