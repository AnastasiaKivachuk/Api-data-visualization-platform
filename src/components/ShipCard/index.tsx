import * as React from 'react';
import {useRouter} from "next/router";

import {Card as MuiCard} from '@mui/material';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

import {ShipsInterface} from "../../interfaces/home.interface";

 const ShipCard = ({image, name, url, year_built, model}: ShipsInterface) => {
    const router = useRouter();
    return (
        <MuiCard sx={{maxWidth: 345}}>
            {image && <CardMedia
                component="img"
                height="140"
                image={image}
                alt="green iguana"
            />}
            <CardContent>
                <Typography gutterBottom variant="h5" component="div">
                    {name}
                </Typography>
                <Typography variant="body2" color="text.secondary">
                    {model} - {year_built}
                </Typography>
            </CardContent>
            {url && <CardActions>
                <Button variant="outlined" size="small" onClick={() => router.push(url)}>Learn More</Button>
            </CardActions>
            }
        </MuiCard>
    );
}

export default ShipCard;