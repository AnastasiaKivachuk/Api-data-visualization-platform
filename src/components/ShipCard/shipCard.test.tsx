import {shallow} from "enzyme";
import React from "react";

import {ShipsInterface} from "../../interfaces/home.interface";
import {HOME_ROUTE} from "../../constants/routes.constants";
import ShipCard from "./index";

const MockShipData: ShipsInterface = {
    image: 'https://i.imgur.com/GvtuYUN.jpg',
    model: 'Marmac 303',
    name: 'NRC Quest',
    url: 'https://www.marinetraffic.com/en/ais/details/ships/shipid:447294/vessel:NRC%20QUEST',
    year_built: 2002
};

const setUp = (props: ShipsInterface) => shallow(<ShipCard {...props} />);


jest.mock('next/router', () => ({
    useRouter() {
        return ({
            route: '/',
            pathname: '',
            query: '',
            asPath: '',
            push: jest.fn(),
            events: {
                on: jest.fn(),
                off: jest.fn()
            },
            beforePopState: jest.fn(() => null),
            prefetch: jest.fn(() => null)
        });
    },
}));

describe("ShipCard component", () => {

    beforeEach(() => {
        const useRouter = jest.spyOn(require("next/router"), "useRouter");
        useRouter.mockImplementation(() => ({
            route: HOME_ROUTE,
            pathname: '',
            query: '',
            asPath: '',
            push: jest.fn(),
            events: {
                on: jest.fn(),
                off: jest.fn()
            },
            beforePopState: jest.fn(() => null),
            prefetch: jest.fn(() => null)
        }));
    })

    it("should render ShipCard component", () => {
        const component = setUp(MockShipData)
        expect(component).toMatchSnapshot();
    })

    it("check all elements with props", () => {
        const component = setUp(MockShipData)
        const text = ['NRC Quest', "Marmac 303 - 2002"]
        expect(component.find('ForwardRef(Typography)').length).toBe(2);
        component.find('ForwardRef(Typography)').map((item, index) => {
            expect(item.text()).toBe(text[index]);
        })
    })

    it("check button link", () => {
        const component = setUp(MockShipData)
        expect(component.find('ForwardRef(Button)').length).toBe(1);
        component.find('ForwardRef(Button)').simulate('click')
    })
})
