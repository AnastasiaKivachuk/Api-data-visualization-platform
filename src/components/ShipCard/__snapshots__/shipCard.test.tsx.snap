// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipCard component should render ShipCard component 1`] = `
<ForwardRef(Card)
  sx={
    Object {
      "maxWidth": 345,
    }
  }
>
  <ForwardRef(CardMedia)
    alt="green iguana"
    component="img"
    height="140"
    image="https://i.imgur.com/GvtuYUN.jpg"
  />
  <ForwardRef(CardContent)>
    <ForwardRef(Typography)
      component="div"
      gutterBottom={true}
      variant="h5"
    >
      NRC Quest
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      color="text.secondary"
      variant="body2"
    >
      Marmac 303
       - 
      2002
    </ForwardRef(Typography)>
  </ForwardRef(CardContent)>
  <ForwardRef(CardActions)>
    <ForwardRef(Button)
      onClick={[Function]}
      size="small"
      variant="outlined"
    >
      Learn More
    </ForwardRef(Button)>
  </ForwardRef(CardActions)>
</ForwardRef(Card)>
`;
